### PR TITLE
Delete notes

### PR DESCRIPTION
**Pull request by omjadhav85**
Note: Please only review the last commit

https://github.com/bLAckBLAZER/my-notes-app/pull/5/commits/c847d7808f48078f67ddfc53dbb53f48c9a32d6f

This PR also contains changes of previous PRs.

Features added in this PR:

1. User can now move the nots to trash
2. User can permanently delete the notes from trash
3. User can restore the notes from trash

### CODE REVIEW

File: src/components/Loaders/Loader.css [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  transform: translate(-50%, -50%);
+}
+
+.loader-container {
+  position: absolute;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  background-color: rgba(0, 0, 0, 0.1);
```
Comment (tusharkandpal) [inline src/components/Loaders/Loader.css:14]: better to make color variables in root.

File: src/components/NavBar/SideBar.css [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  justify-content: flex-end;
+  border-bottom: none;
+}
+
+.mob-aside .side-bar-item a {
+  text-decoration: none;
+}
+/* .mob-aside .side-bar-item:last-child {
+  justify-self: flex-end;
+} */
```
Comment (tusharkandpal) [inline src/components/NavBar/SideBar.css]: please remove this commented part.

File: src/components/Note/Note.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+              onClick={() =>
+                updateNote(
+                  { ...note, isPinned: !note.isPinned },
+                  dispatchData,
+                  setIsLoading,
+                  token
+                )
+              }
+            >
+              <MdPushPin size={20} color={note.isPinned ? "00a82d" : "black"} />
```
Comment (tusharkandpal) [inline src/components/Note/Note.js:32]: suggestion: can make color variables or classes for the same.

File: src/components/SideNavBar/SideNavBar.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        <SideNavBarItem itemName="Archive" itemIcon={<MdArchive size={28} />} />
+      </NavLink>
+      <NavLink
+        to="/trash"
+        style={({ isActive }) => {
+          return {
+            backgroundColor: isActive && "#00a82d",
+            color: isActive && "white",
+          };
+        }}
```
Comment (tusharkandpal) [inline src/components/SideNavBar/SideNavBar.js]: can extract this repeated code in a separate function

File: src/utils/noteServerCalls.js [hunk 5]
```diff
@@ -0,0 +1,249 @@
+import axios from "axios";
+// import { isPresentInList } from "./helperFunctions";
```
Comment (tusharkandpal) [inline src/utils/noteServerCalls.js]: please remove this if not needed.

File: src/utils/noteServerCalls.js [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+      },
+    });
+
+    if (res.status === 200) {
+      dispatchData({ type: "SET_ALL_NOTES", payload: res.data.notes });
+    } else {
+      console.error("get all notes call failed with status: ", res.status);
+    }
+  } catch (err) {
+    console.error("Error getting all notes", err);
```
Comment (tusharkandpal) [inline src/utils/noteServerCalls.js:21]: here also you can use `toast`

**Review by tusharkandpal — APPROVED** [review]
Very clean code & well implemented. Good work 💯

**Review by kedark152 — APPROVED** [review]
Clean Code, Good use of variable names.

**Pull request merged**